torch/onnx/symbolic_opset9.py

```diff
@@ -2248,33 +2248,58 @@ def topk(g, self, k, dim, largest, sorted, out=None):
 
 
 def to(g, self, *args):
-    # ONNX doesn't have a concept of a device, so we ignore device casts
-    if len(args) == 4:
-        if args[0].node().kind() == "prim::device" or args[0].type().isSubtypeOf(ListType.ofInts()):
+
+    def is_aten_to_device_only(*args):
+        if len(args) == 4:
             # aten::to(Tensor, Device, bool, bool, memory_format)
-            return self
+            if args[0].node().kind() == "prim::device" or \
+                    args[0].type().isSubtypeOf(ListType.ofInts()):
+                return True
+            if sym_help._is_value(args[0]) and \
+                    args[0].node().kind() == "onnx::Constant" and \
+                    isinstance(args[0].node()["value"], str):
+                return True
+        elif len(args) == 5:
+            # aten::to(Tensor, Device, ScalarType, bool, bool, memory_format)
+            dtype = sym_help._get_const(args[1], "i", "dtype")
+            if not dtype:
+                # When dtype is None, this is a aten::to(device) call
+                return True
+        elif len(args) >= 6 and len(args) <= 7:
+            # aten::to(Tensor, ScalarType, Layout, Device, bool, bool, memory_format) -> Tensor
+            # aten::to(Tensor, ScalarType, Layout, Device, bool, bool, bool, memory_format) -> Tensor
+            dtype = sym_help._get_const(args[0], "i", "dtype")
+            if not dtype:
+                # When dtype is None, this is a aten::to(device) call
+                return True
+        return False
+
+    # ONNX doesn't have a concept of a device, so we ignore device-only casts
+    if is_aten_to_device_only(*args):
+        return self
+
+    if len(args) == 4:
+        # TestONNXRuntime::test_ones_bool shows args[0] of aten::to() can be onnx::Constant[value=<Tensor>]()
+        # In this case, the constant value is a tensor not int,
+        # so sym_help._maybe_get_const(args[0], 'i') would not work.
+        dtype = args[0]
+        if sym_help._is_value(args[0]) and args[0].node().kind() == "onnx::Constant":
+            tval = args[0].node()["value"]
+            if isinstance(tval, torch.Tensor):
+                if len(tval.shape) == 0:
+                    tval = tval.item()
+                    dtype = int(tval)
+                else:
+                    dtype = tval
```

Comment:
what does a non-scalar dtype mean?

Reply:
Not sure, this is also pre-existing code that handles `to(dtype)` scenarios, whereas this PR focus on `to(device)` only.

The code implies that there are two ways of setting `dtype` based on a `Tensor`: using a single int or a list of them, but looking at [doc](https://pytorch.org/docs/master/generated/torch.Tensor.to.html?highlight=#torch.Tensor.to) there is no such scenario

Reply:
Sorry I didn't realize this wasn't changed. Let's leave it for now.